### ISSUE TITLE
🐛 BUG: 

### ISSUE THREAD

**Issue by dgavey** [withastro/astro#3480]
### What version of `astro` are you using?

v1.0.0-beta.36

### Are you using an SSR adapter? If so, which one?

Netlify

### What package manager are you using?

pnpm

### What operating system are you using?

mac

### Describe the Bug

Not sure if it's a bug or just something I'm missing.  But I can't seem to find the query parameters as part of a GET api call. 

Attached a reproduction with more details.  

From what I can see even in the internals of the request object, the query params are stripped.  Would be very useful if they were either part of the "params" or accessible through the request object.

Example file and call 

```

// location: pages/api/testqp.ts
import type { APIContext } from 'astro';

export async function get({ params, request }: APIContext) {
  console.log('params', params);
  console.log('request', request);
  // Calling /api/testqp?myparam=1&param2=test
  // Query Params of the get api request do not appear anywhere in the request object or the params
  // All QP's seem to have been stripped

  return new Response(JSON.stringify({ param: params }), {
    status: 200,
    headers: {
      'Content-Type': 'application/json',
    },
  });
}

```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-vwbgay?file=src/pages/api/testqp.ts

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by dgavey** [comment]
Oddly after coming back to this, I'm seeing it locally now, but the example still doesn't show.

**Comment by dgavey** [comment]
Ah, unless there is an integration for a deployment service aka netlify (in my case) then the query params are stripped from the request.url

**Comment by matthewp** [comment]
Yep, that's correct, because in SSG mode there are no query params to look at, that's only available on SSR mode. Closing as not a bug.

**Comment by dgavey** [comment]
I'd suggest maybe some of error or documentation update on it to help devs in the future.  Would be happy to add a PR for something like that if you have a suggestion.

**Comment by jsadoski-rockhall** [comment]
Sorry to comment on an old closed issue, but this is the top result on Google.

As long as you intend on using SSR, simply setting `output: "server"` in your `astro.config.mjs` will make the query parameters available for API endpoints.

Reference: [Configuration Reference: Output](https://docs.astro.build/en/reference/configuration-reference/#output)
